### PR TITLE
feat/event_search

### PR DESCRIPTION
**Pull request by nj606214**

### CODE REVIEW

File: components/EventsColumn.tsx [hunk 1]
```diff
@@ -281,15 +281,110 @@ export default function EventsColumn() {
...
+    const params = new URLSearchParams({
+      searchQuery: searchInput
+    });
+
+    const response = await fetch(`/api/events/search?${params.toString()}`, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
```
Comment (raydelvecchio) [inline components/EventsColumn.tsx:348]: Receiving the following error when I make the API call by submitting a query and pressing the button on the frontend:

`SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (node:internal/deps/undici/undici:6662:19)
    at successSteps (node:internal/deps/undici/undici:6636:27)
    at node:internal/deps/undici/undici:1236:60
    at node:internal/process/task_queues:140:7
    at AsyncResource.runInAsyncScope (node:async_hooks:203:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:137:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)`

File: components/EventsColumn.tsx [hunk 2]
```diff
@@ -281,15 +281,110 @@ export default function EventsColumn() {
...
+  }
+  type Match = {
+    document: EventDocument
+  }
+  
+  type SearchResponse = {
+    matches: Match[]
+    num_matches: number
+    search_time_ms: number
+  }
```
Comment (iporollo) [inline components/EventsColumn.tsx:309]: types and interface definitions should be outside of components

File: components/EventsColumn.tsx [hunk 3]
```diff
@@ -281,15 +281,110 @@ export default function EventsColumn() {
...
+        type: event.document.type,
+      }
+      mappedEvents.push(mappedEventObj)
+    })
+    return mappedEvents
+  }
+
+  const fetchSearchedEvents = async () => {
+    if (searchInput.length < 1) {
+      console.error('no search input');
```
Comment (iporollo) [inline components/EventsColumn.tsx:332]: console.error not necessary here

**Review by iporollo — CHANGES_REQUESTED** [review]

File: components/EventsColumn.tsx (hunk 1 above)
Comment (nj606214) [reply components/EventsColumn.tsx:348]: hmm not seeing that on my end, do you have the latest updates from feat/event_search_endpoint on platform-api running? heres a link to that pr https://github.com/cerebralvalley/platform-api/pull/51

**Pull request closed without merging**